airflow-core/src/airflow/serialization/serialized_objects.py

```
@@ -925,8 +925,11 @@ def deserialize(cls, encoded_var: Any) -> Any:
         elif type_ == DAT.DATETIME:
             return from_timestamp(var)
         elif type_ == DAT.POD:
-            if not _has_kubernetes():
-                raise RuntimeError("Cannot deserialize POD objects without kubernetes libraries installed!")
+            # lazy import PogGenerator here if needed and set HAS_KUBERNETES to True and import the generator
+            # because if we need to deserialize Pod, we MUST have PodGenerator imported
+            _has_kubernetes(force=True)
+            from airflow.providers.cncf.kubernetes.pod_generator import PodGenerator
+
```

@potiuk This is a wrong fix! and will result in Scheduler or API-server crashing if k8s library was uninstalled/not-present anymore with weird errors

---

```suggestion
            if not _has_kubernetes(force=True):
                raise RuntimeError("Cannot deserialize POD objects without kubernetes libraries installed!")

```

---

It will already be imported by `_has_kubernetes` if k8s provider and library is installed after your "force" flag to recheck

---

I see - but Right now it crashes if the other side has no k8s libraries installed (see main failures) - because it cannot deserialize the objects - so I guess we should simply silently ignore it

---

Should be beter now.

---

The `force=True` alone will be enough without any changes. I had pushed a fix on your PR :) before your force-pushed it too :)